intro-to-R-tidyverse/00b-debugging_resources.md

```diff
@@ -0,0 +1,244 @@
+that you believe should be in your environment, are in your environment. 
+
+It's also good practice to be refreshing your R session and clearing your R 
+notebook output. If you are encountering problems and haven't refreshed your
+R session, you may want to do that before further troubleshooting. 
+
+### 4) Google your error message
+
+The main advantage to Googling your errors, is that you likely not the first 
+person to encounter the problem. Certain phrases and terms in the error message
+will yield more fruitful search results then others.  
+
+When you do Google, a few common sources that will probably come up that we 
+recommend looking at are:
+
+#### a) [StackOverflow](https://stackoverflow.com/)
+StackOverflow this is a forum where people post questions and problems they 
+encounter in their code. 
+
+#### b) [GitHub Issues](https://help.github.com/en/articles/about-issues)
+People also will post their problems to GitHub issues. Often these are more
+geared toward fixing problems with the package or software itself, but this is 
+a way to potentially get direct help on an issue from the authors of the package
+you are using.
+
+#### c) [R-bloggers](https://www.r-bloggers.com/)
+R-bloggers has examples of R code that you can use to figure out how to construct
+various analyses. This is a good resource for example code, although it's format
+isn't built for asking exact questions like StackOverflow.
+
+### 5) Look at the documentation for a function to make sure you are using it correctly
+
+Once you've better determined the origin of the problem, you should use whatever
+documentation is available to you regarding the problematic code. When using 
+a new function from a package you are unfamiliar with, it's worthwhile to skim
+through the documentation so you know how to properly use the functions. For 
+base R functions, Tidyverse functions, and *some* Bioconductor packages, the 
+documentation will give you a lot of the information you need. However, you will 
+also likely find that not all documentation is thorough or clear. 
+
+As we discussed in 
+[`intro_to_R` module](https://alexslemonade.github.io/training-modules/intro-to-R-tidyverse/01-intro_to_r.nb.html),
+objects have *structures* and *types*. Having input that doesn't match the 
+requirements that a function has can be a common source of errors. Pay special
+attention to what the documentation says about what kind of input and output 
+the function is designed to use. 
+
+#### Use the RStudio help bar
+
+Here's a screenshot from the help window in RStudio. Note that here we searched
+for the `levels` function. R documentation includes information about what the
+expected arguments are as well as examples of how to use a function. Note here 
+that this documentation tells us that the input for `x` is probably a factor. 
+![search_bar](screenshots/r_search_bar.png)
+
+#### For Bioconductor package functions, look at their documents
```

Comment:
Should we mention that many have _vignettes_ in addition to the Reference Manual which I believe is what you are calling out here?

Reply:
I might change this to
```suggestion
#### For Bioconductor package functions, look at their page on bioconductor.org
```
then